src/software/new_geom/linear_spline_test.cpp

```diff
@@ -0,0 +1,197 @@
+#include <gtest/gtest.h>
+
+#include "software/new_geom/linear_spline2d.h"
+
+TEST(LinearSplineTest, test_three_knot_value_at)
+{
+    std::vector<Point> points({Point(4, 18), Point(3, 0), Point(-7, -1)});
+    LinearSpline2d s(points);
+    EXPECT_EQ(s.getNumKnots(), points.size());
```

Comment:
nit: technically checking `NumKnots` is out of scope of this unit test (according to the name). Consider breaking it up

Reply:
After offline discussion we decided to leave as is, but rename to indicate we're effectively testing the constructor.